packages/core/svelte/src/walletStore.ts

```diff
@@ -151,6 +153,7 @@ async function disconnect(): Promise<void> {
             ...storeValues,
             disconnecting: true,
         }));
+        await destroyAdapter();
```

Comment:
`destroyAdapter` is not an async function.

I think we may want to rename it as well (it doesn't destroy the adapter, or anything really). Maybe just a generic `cleanup` function would do.